### DISCUSSION TITLE
How to add user as my project register user not as admin user to allow commenting for post..

### DISCUSSION THREAD

**Discussion by SALIMA534** [Radi85/Comment#188] [Q&A]
Hi , I have integrated django-comments-dab package into my project. Features of django-comments-dab package is but How to use user as our project register as not as admin user it is considering as admin user how to override this and I want to show user as my project registered users. Please help me to achieve this, Thanks.

**Comment by abhiabhi94** [comment]
I'm afraid I don't understand your query completely.

**Reply by SALIMA534** [reply]
Hi abhi,

I am asking by default based on **admin logged in user** **django-comments-xtd** allowing user to comment. 
How to use user as my site user like In my project all registered users are in **customer**(model in models.py) DB and I want to allow user to comment based on customer model how can i achieve this.

Please help me.

**Reply by abhiabhi94** [reply]
If I understand you correctly, you're asking our integration of Custom user model.

Custom user models are supported. In case you are facing a particular issue, you can point that to us.

**Reply by SALIMA534** [reply]
Thank you abhi, May I know how to integrate my custom user to integrate. Please help me to achieve it.

Actually I have went through whole document but I didn't get where to integrate custom user to allow to post a comment.
I mean after custom user login only user can post a comment for Ad.

If solve this issue it means a lot abhi.

**Reply by abhiabhi94** [reply]
just use it the same way you would even if you weren't using a custom user model. there are no changes required(from the user's side) to integrate it with a custom user model.

**Reply by SALIMA534** [reply]
Hi,

We added customer user model in our project


`AUTH_USER_MODEL = 'pages.vk_customer'
`

```

from django.contrib.auth.models import BaseUserManager, AbstractBaseUser


class UserAccountManager(BaseUserManager):
    def create_user(self, email, firstname, lastname, password=None):
        if not email:
            raise ValueError('Email must be set!')
        user = self.model(email=email, firstname=firstname, lastname=lastname)
        user.set_password(password)
        user.save(using=self._db)
        return user

    def create_superuser(self, email, firstname, lastname, password):
        user = self.create_user(email, firstname, lastname, password)
        user.is_admin = True
        user.save(using=self._db)
        return user

    def get_by_natural_key(self, email_):
        return self.get(email=email_)

```

and we seeing an error like this

`'vk_customer' object has no attribute 'set_password'
`
Could you let us know what we are missing? We really liked your project and would like to integrate into our Web application.

Best regards,
Salima

**Reply by abhiabhi94** [reply]
I'm sorry I couldn't respond earlier due to the pandemic.

The error that you should above(as of now), doesn't seem to be related to `comment` model. Why do you think this issue is caused by the `comment` model?
